src/main/java/com/connexta/store/adaptors/S3StorageAdaptor.java

```diff
@@ -110,6 +121,14 @@ public StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key)
     }
   }
 
+  @Override
+  public void updateStatus(String datasetId, String storeStatus) {
+    final Tag updatedTag = new Tag(STATUS_KEY, storeStatus);
+    final ObjectTagging objectTagging = new ObjectTagging(ImmutableList.of(updatedTag));
+
+    amazonS3.setObjectTagging(new SetObjectTaggingRequest(bucket, datasetId, objectTagging));
```

Comment:
❗️Can this fail? What are the situations that would cause this to fail? I think that should be documented in the javadoc and tested.

Reply:
Yeah it can fail for the same reasons as other instances of AmazonS3: the client is unavailable, the S3 service is unavailable, etc.